pkg/cloud/metadata.go

```diff
@@ -69,13 +75,69 @@ func (m *Metadata) GetOutpostArn() arn.ARN {
 func NewMetadata() (MetadataService, error) {
 	sess := session.Must(session.NewSession(&aws.Config{}))
 	svc := ec2metadata.New(sess)
-	return NewMetadataService(svc)
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return NewMetadataService(svc, clientset)
```

Comment:
IMO we should wrap this error in something helpful. like: error getting information from metadata service or node object : err. Looking back at the issue where people reported that the driver couldnt run because of metadata being unavailable, some context would have helped, because otherwise the driver just crashes with weird error.

Reply:
can you confirm that this is the idiomatic way to wrap errors?
`func NewMetadata() (MetadataService, error) {
	sess := session.Must(session.NewSession(&aws.Config{}))
	svc := ec2metadata.New(sess)
	// creates the in-cluster config
	config, err := rest.InClusterConfig()
	if err != nil && !svc.Available() {
		return nil, err
	}
	// creates the clientset
	clientset, err := kubernetes.NewForConfig(config)
	if err != nil && !svc.Available() {
		return nil, err
	}
	metadataService, err := NewMetadataService(svc, clientset)
	if err != nil {
		err = fmt.Errorf("error getting information from metadata service or node object: %w", err)
	}
	return metadataService, err
}`